### PR TITLE
Replace sodium-native with @liskhq/lisk-cryptography - Closes #3426

### LINKED ISSUE #3426: Replace sodium-native with @liskhq/lisk-cryptography
### Expected behavior
There should be one library wrapping `libsodium` used in Lisk SDK.

### Actual behavior
There are two libraries wrapping `libsodium` used in Lisk SDK currently:
"@liskhq/lisk-cryptography" and "sodium-native".

Both of them expose a similar API.

### Which version(s) does this affect? (Environment, OS, etc...)
`1.6.0`

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
Framework was not using cryptography library

### How did I fix it?
Consistently use lisk-cryptography library

### How to test it?
npm t and run devnet

### Review checklist

* The PR resolves #3426 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/logic/block.js [hunk 1]
```diff
@@ -106,11 +109,13 @@ class Block {
 		let size = 0;
 
 		const blockTransactions = [];
-		const payloadHash = crypto.createHash('sha256');
+		const payloadHash = deepHashBuffer(
```
Comment (ManuGowda) [inline framework/src/modules/chain/logic/block.js]: const transactionsBytes = transactions.map(transaction => transaction.getBytes());
const payloadHash = deepHashBuffer(transactionsBytes);

File: framework/src/modules/chain/logic/block.js [hunk 2]
```diff
@@ -106,11 +109,13 @@ class Block {
...
+			transactions.map(transaction => transaction.getBytes())
+		);
 
 		for (let i = 0; i < transactions.length; i++) {
 			const transaction = transactions[i];
-			const bytes = transaction.getBytes(transaction);
+			const bytes = transaction.getBytes();
```
Comment (ManuGowda) [inline framework/src/modules/chain/logic/block.js]: const bytes = transactionsBytes[i];

File: framework/src/modules/chain/logic/block.js [hunk 3]
```diff
@@ -265,10 +251,7 @@ class Block {
  * @todo Add description for the params
  */
 __private.getAddressByPublicKey = function(publicKey) {
-	const publicKeyHash = crypto
-		.createHash('sha256')
-		.update(publicKey, 'hex')
-		.digest();
+	const publicKeyHash = hash(publicKey, 'hex');
```
Comment (ManuGowda) [inline framework/src/modules/chain/logic/block.js]: `hashedPublicKey`? that way its more consistent with others like `hashedBlock`, `hashedData`

File: framework/src/modules/chain/logic/block.js (hunk 1 above)
Comment (shuse2) [reply framework/src/modules/chain/logic/block.js]: why separate into 2 parts?

File: framework/src/modules/chain/logic/block.js (hunk 2 above)
Comment (shuse2) [reply framework/src/modules/chain/logic/block.js]: probably this part needs to be cleaned up in other issue

File: elements/lisk-cryptography/src/hash.ts [hunk 4]
```diff
@@ -43,3 +43,14 @@ export const hash = (data: Buffer | string, format?: string): Buffer => {
 		'Unsupported data format. Currently only Buffers or `hex` and `utf8` strings are supported.',
 	);
 };
+
```
[the remark this answers is not in the document]
Comment (shuse2) [reply elements/lisk-cryptography/src/hash.ts]: We are not using `JSDocs` for typescripts. It already tells what the input is.
Also, this hash function is not related to endianness

**Comment by shuse2** [comment]
Closing due to big changes, and need to reimplement

**Pull request closed without merging**